modules/samples/connectedcup/component/api/src/main/java/org/coffeeking/api/ConnectedCupControllerServiceImpl.java

```diff
@@ -20,73 +20,98 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.coffeeking.api.util.APIUtil;
+import org.coffeeking.api.util.SensorRecord;
 import org.coffeeking.connectedcup.plugin.constants.ConnectedCupConstants;
-import org.coffeeking.api.transport.ConnectedCupMQTTConnector;
-import org.wso2.carbon.device.mgt.iot.controlqueue.mqtt.MqttConfig;
-import org.wso2.carbon.device.mgt.iot.exception.DeviceControllerException;
-import org.wso2.carbon.device.mgt.iot.service.IoTServerStartupListener;
-import org.wso2.carbon.device.mgt.iot.transport.TransportHandlerException;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.authorization.DeviceAccessAuthorizationException;
+import org.wso2.carbon.analytics.dataservice.commons.SORT;
+import org.wso2.carbon.analytics.dataservice.commons.SortByField;
+import org.wso2.carbon.analytics.datasource.commons.exception.AnalyticsException;
 
+import javax.ws.rs.Consumes;
 import javax.ws.rs.GET;
-import javax.ws.rs.HeaderParam;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
 
 public class ConnectedCupControllerServiceImpl implements ConnectedCupControllerService {
 
     private static Log log = LogFactory.getLog(ConnectedCupControllerServiceImpl.class);
-    private static ConnectedCupMQTTConnector connectedCupMQTTConnector;
 
     @Path("device/ordercoffee")
     @POST
-    public Response orderCoffee(@QueryParam("deviceId") String deviceId, @QueryParam("deviceOwner") String deviceOwner) {
-        log.info("Coffee ordered....!");
-
-        if (log.isDebugEnabled()) {
-            log.debug("Sending request to read liquid level value of device [" + deviceId + "] via MQTT");
+    public Response orderCoffee(@QueryParam("deviceId") String deviceId) {
+        try {
+            if (!APIUtil.getDeviceAccessAuthorizationService().isUserAuthorized(new DeviceIdentifier(deviceId,
+                                                 ConnectedCupConstants.DEVICE_TYPE))) {
+                return Response.status(Response.Status.UNAUTHORIZED.getStatusCode()).build();
+            }
+            log.info("Coffee ordered....!");
+            if (log.isDebugEnabled()) {
+                log.debug("Sending request to read liquid level value of device [" + deviceId + "] via MQTT");
+            }
+            return Response.ok().entity("Coffee ordered.").build();
+        } catch (DeviceAccessAuthorizationException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
```

Comment:
This exception is thrown when there is an error occurred when calling the authorization service,eg: plugin or group management initialization failure. So this would raise due to an internal issue. So isn't the response Internal Server Error is the correct one ?